doc/guides/arch/arm_cortex_m.rst

```diff
@@ -0,0 +1,643 @@
+.. _arm_cortex_m_developer_guide:
+
+Arm Cortex-M Developer Guide
+############################
+
+Overview
+********
+
+This page contains detailed information about the status of the Arm Cortex-M
+architecture porting in the Zephyr RTOS and describes key aspects when
+developing Zephyr applications for Arm Cortex-M-based platforms.
+
+Key supported features
+**********************
+
+The table below summarizes the status of key OS features in the different
+Arm Cortex-M implementation variants.
+
+
++---------------------------------+-----------------------------------+-----------------+---------+--------+-----------+--------+---------+------------+------------+
```

Comment:
`.. list-table::` may lead to more readable table, see e.g. https://github.com/zephyrproject-rtos/zephyr/blob/main/doc/contribute/coding_guidelines/index.rst

Reply:
The problem with this table design is that it is not possible to have mixed cell configuration, e.g. merged rows and columns arbitrarily (or at least I have not found an easy way to do it).
The tables in this guide use heavily rows and columns of variable sizes, so I am a bit reluctant to switch to using a more restrictive table framework.